nemo/core/nemo_model.py

```diff
@@ -0,0 +1,240 @@
+                            d_format=DeploymentFormat.TRTONNX,
+                        )
+                    except Exception as ex:
+                        print(ex)
+                        logging.warning(f"Did not convert {module_name} to .onnx")
+                        module_checkpoint = module_name + ".pt"
+                        module.save_to(path.join(tmp_folder, module_checkpoint))
+                else:
+                    module_checkpoint = module_name + ".pt"
+                    module.save_to(path.join(tmp_folder, module_checkpoint))
+
+            make_nemo_file_from_folder(resulting_file, tmp_folder)
+            logging.info(f"Exported model {self} to {resulting_file}")
+        except:
+            logging.error("Could not perform NeMoModel export")
+        finally:
+            shutil.rmtree(rnd_path)
+            pass
+
+    @property
+    @abstractmethod
+    def modules(self) -> Iterable[NeuralModule]:
+        pass
+
+    @property
+    @abstractmethod
+    def train_graph(self) -> NeuralGraph:
+        pass
+
+    @property
+    @abstractmethod
+    def eval_graph(self) -> NeuralGraph:
+        pass
+
+    def train(self):
+        """
+        Sets model to the training mode
+
+        Returns:
+            None
+        """
+        self._operation_mode = OperationMode.training
+        for module in self.modules:
+            module.operation_mode = OperationMode.training
+            if module.type == ModuleType.trainable and hasattr(module, 'train'):
+                module.train()
+
+    def eval(self):
+        """
+        Sets model to the evaluation mode
+
+        Returns:
+            None
+        """
+        self._operation_mode = OperationMode.evaluation
+        for module in self.modules:
+            module.operation_mode = OperationMode.evaluation
+            if module.type == ModuleType.trainable and hasattr(module, 'eval'):
+                module.eval()
+
+    @staticmethod
+    def __unpack_nemo_file(path2file: str, out_folder: str = None) -> str:
+        if not path.exists(path2file):
+            raise FileNotFoundError(f"{path2file} does not exist")
+        if out_folder is None:
+            out_folder = ''.join(
+                random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(16)
+            )
+
+        tar = tarfile.open(path2file, "r:gz")
+        tar.extractall(path=out_folder)
+        tar.close()
+        return path.join(out_folder, NEMO_TMP_FOLDER), out_folder
+
+    def get_weights(self):
+        raise NotImplemented()
+
+    def set_weights(
+        self, name2weight, name2name_and_transform,
+    ):
+        raise NotImplemented()
+
+    def tie_weights_with(
+        self, module, weight_names, name2name_and_transform,
+    ):
+        raise NotImplemented()
+
+    def restore_from(self, path: str):
+        raise NotImplemented("Please use from_pretrained method for NeMoModels")
```

Comment:
how about simply calling from_pretrained()?

or the arguments won't match...?

Reply:
args won't match. Also restore_from is a member method which assumes that object is already created and just loads weights. "from_pretrained" is a class method which actually creates an object